docs/guides/creatingasource.md

```diff
@@ -0,0 +1,118 @@
+# Creating a Source
+
+In this guide we will create a point to point bridge between an AWS SQS queue and a microservice application called `sockeye` which displays events in web interface. This simple flow is depicted below.
+
+![](../assets/images/sqs-sockeye.png)
+
+We will create:
+
+- [x] The sockeye target which serves as an event display.
+- [x] The `AWSSQSSource` which consumes events from an AWS SQS queue.
+
+## Event display
+
+First of all, we need to have a tool to see the events that come from our source.
+
+Create a `sockeye` service by saving the following YAML manifest in a file called `sockeye.yaml` and applying it to your Kubernetes cluster:
+
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: sockeye
+spec:
+  template:
+    spec:
+      containers:
+        - image: docker.io/n3wscott/sockeye:v0.7.0@sha256:e603d8494eeacce966e57f8f508e4c4f6bebc71d095e3f5a0a1abaf42c5f0e48
+```
+
+```
+kubectl apply -f sockeye.yaml
+```
+
+Open the web interface in a browser:
+
+```shell
+browse $(kubectl get ksvc sockeye -o=jsonpath='{.status.url}')
+```
+
+## Create a AWS SQS Event source
+
+You can explore the specification of the object using the `kubectl explain` command. You will see that you need the ARN (i.e Amazon Resource Name) or your AWS SQS queue and you need your AWS API keys.
+
+
+```console
+$ kubectl explain awssqssource.spec
+KIND:     AWSSQSSource
+VERSION:  sources.triggermesh.io/v1alpha1
+
+RESOURCE: spec <Object>
+
+DESCRIPTION:
+     Desired state of the event source.
+
+FIELDS:
+   arn	<string> -required-
+     ARN of the Amazon SQS queue to consume messages from. The expected format
+     is documented at
+     https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonsqs.html#amazonsqs-resources-for-iam-policies.
+
+   credentials	<Object>
+     Credentials to interact with the Amazon SQS API. For more information about
+     AWS security credentials, please refer to the AWS General Reference at
+     https://docs.aws.amazon.com/general/latest/gr/aws-security-credentials.html
+
+   receiveOptions	<Object>
+     Options that control the behavior of message receivers.
+
+   sink	<Object> -required-
+     The destination of events sourced from Amazon SQS.
+```
+
+```yaml
+apiVersion: sources.triggermesh.io/v1alpha1
+kind: AWSSQSSource
+metadata:
+  name: sqs-guide
+spec:
+  arn: arn:aws:sqs:us-east-1:587264312345:triggermesh
```

Comment:
We should avoid real ARNs in public facing docs. AWS uses `123456789012` in their own docs.